observer/observer.go

```diff
@@ -202,6 +204,58 @@ func (o *Observer) IsAnyJobRunning(repository string) bool {
 	return false
 }
 
+// HasLimitConcurrentJobsReached checks if the limit of concurrent jobs by type (backup, check, etc)
+// has been reached
+func (o *Observer) HasLimitConcurrentJobsReached(jobType, repository string) bool {
+	listOfJobs := o.GetJobsByRepository(repository)
+	fmt.Println("type: ", jobType)
+	switch jobType {
+	case "backup":
+		l := cfg.Config.GlobalConcurrentArchiveJobsLimit
+		return filterByJobType(l, jobType, listOfJobs)
+	case "check":
+		l := cfg.Config.GlobalConcurrentCheckJobsLimit
+		return filterByJobType(l, jobType, listOfJobs)
+	case "archive":
+		l := cfg.Config.GlobalConcurrentArchiveJobsLimit
+		return filterByJobType(l, jobType, listOfJobs)
+	case "prune":
+		l := cfg.Config.GlobalConcurrentPruneJobsLimit
+		return filterByJobType(l, jobType, listOfJobs)
+	case "restore":
+		l := cfg.Config.GlobalConcurrentRestoreJobsLimit
+		return filterByJobType(l, jobType, listOfJobs)
+	default:
+		// In case of type "" -> Job
+		return false
+	}
```

Comment:
All job implement the Executor interface. Does it make sense to add a Getter there for accessing the limit? This would eliminate this switch-case with very similar code.
@Kidswiss ?

Reply:
@ccremer in the Getter, I should pass the job type to return the proper limit value, right?

Reply:
I thought something about this:
```go
$ cat queue/execution.go
type Executor interface {
	...
	GetConcurrencyLimit() int
}

$ cat executor/restore.go
func (r *RestoreExecutor) GetConcurrencyLimit() int {
	return cfg.Config.GlobalConcurrentRestoreJobsLimit
}
```
And return 0 or -1 if no limit set or supported. Or the function returns an error, that would make it explicit (I regard returning 0 or -1 sometimes as lazyness when something doesn't feel right regarding limits)

Reply:
you're completely right, thanks